array_methods_lab.js

```diff
@@ -0,0 +1,341 @@
+// # Exercises
+//
+console.log(`1. Given an array of numbers, write a function that returns
+  a new array where every element is incremented by 10.`);
+//
+// ```
+let inTenOut = [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+let byTen = inTenOut.map(elem => elem +=10);
+console.log(byTen);
+
+console.log(`\n2. Given an array, return a new array where every
+  non-string element is converted to an empty string.`);
+//
+// ```
+let strOrNot = ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+let testing = strOrNot.map(el => {
+  if (typeof el != 'string'){
+    return el ='';
+  } else {
+    return el;
+  }
+});
+//logging the new array
+ console.log(testing);
+
+
+console.log(`\n3. Write a function that returns only the even
+  elements in an array of numbers.`);
+//
+// ```
+let eveOut = [1,2,3,4,5,10,20]
+// Output: [2,4]
+// ```
+
+const evenFunc =(evenNum) =>{
+return evenNum.filter(el => el % 2 === 0);
+}
+console.log(evenFunc(eveOut));
+
+console.log(`\n4. Write a function that checks if all elements in the array are the same.`);
+//
+// ```
+let sameArr = [1,1,1,1,2];
+// Output: true
+//
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+const multfunc =(same) =>{
+  let num = same[0]
+  let test2 = same.every(el => {
+    if (el === num){
+      // console.log(el);
+      return true;
+    }
+      //console.log(el);
+      return false;
+  });
+  return test2;
+};
+console.log(multfunc(sameArr));
+
```

Comment:
Remember that the every method returns a boolean based on if the condition is true or not. So the return true; return false isn't necessary. Instead we can return the condition of el === num .

Reply:
const multfunc =(same) =>{
	  let num = same[0]
	  return test2 = same.every(el => {
	    return (el === num){
	  });
	};